praetorian_cli/sdk/entities/schedules.py

```
@@ -0,0 +1,253 @@
+    """The methods in this class are to be accessed from sdk.schedules, where sdk
+    is an instance of Chariot."""
+
+    def __init__(self, api):
+        self.api = api
+
+    @staticmethod
+    def _check_response(resp):
+        """Check HTTP response and raise on failure, matching SDK's process_failure pattern."""
+        if not resp.ok:
+            message = f'[{resp.status_code}] Request failed' + (f'\nError: {resp.text}' if resp.text else '')
+            raise Exception(message)
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)